patches/tModLoader/Terraria/ModLoader/GlobalItem.cs

```diff
@@ -14,15 +14,14 @@ namespace Terraria.ModLoader
 	/// </summary>
 	public abstract class GlobalItem : ModType
 	{
-		internal int index;
-		internal int instanceIndex;
+		internal short index;
 
 		protected sealed override void Register() {
 			ItemLoader.VerifyGlobalItem(this);
 
 			ModTypeLookup<GlobalItem>.Register(this);
 			
-			index = ItemLoader.globalItems.Count;
+			index = (short)ItemLoader.globalItems.Count;
```

Comment:
need an exception if item count ever passes a `short`
Also use `ushort` everywhere for indices